### DISCUSSION TITLE
Indented subtasks

### DISCUSSION THREAD

**Discussion by Fractal147** [todotxt/todo.txt#39] [Show and tell]
Thought I'd discuss my own variation on todo.txt
I found the @context and +project tags to be a bit slow to work with, annoying when dealing with many small subtasks.
I also didn't use any of the normal clients for sorting, so was free to mess around with line structure.
Being text, I decided on a new rule-if a line appears below another in the input file and has a greater number of indents at the start, it is a subtask.
So you essentially create a tree of tasks and subtasks branching off.

Sorting has to be aware (sorting recursively with the least indented tasks being the most important) but is otherwise the same.
[tabtodoview](https://github.com/Fractal147/tabtodoview) is my implementation of this - it's processor script that takes the input text file and sorts it in a tree aware way into a text file for display.

    (C) Example Main task 1
    	Example subtask 1a
    	(A) Example subtask 1b
    		(Z) Example subsubtask 1ba
    		(E) Example subsubtask 1bb
    	(F) Example subtask 1c
    (A) Example main task 2

becomes:

    (A) Example main task 2
    (C) Example Main task 1
    	(A) Example subtask 1b
    		(E) Example subsubtask 1bb
    		(Z) Example subsubtask 1ba
    	(F) Example subtask 1c
    	Example subtask 1a

I expect this breaks some paradigms of todo.txt - but hey, this is a discussion.

**Comment by clach04** [comment]
@Fractal147 you might also be interested in some of the todo.md versions, that use GFM task lists https://github.blog/2014-04-28-task-lists-in-all-markdown-documents/

It's currently more fractured than todo.txt and very few implementations/tools

* https://github.com/todomd/todo.md
* https://github.com/todo-md/todo-md/
